src/main/java/io/netty/buffer/api/ByteCursor.java

```diff
@@ -17,48 +17,55 @@
 import io.netty.util.ByteProcessor;
 
 /**
- * The ByteIterator scans through a sequence of bytes.
+ * The ByteCursor scans through a sequence of bytes.
  * This is similar to {@link ByteProcessor}, but for external iteration rather than internal iteration.
  * The external iteration allows the callers to control the pace of the iteration.
  * The API includes methods for reading {@code long}s as a batch of 8 bytes.
  * The long values are always in big-endian format, so that the highest-order byte in the long value, contain the byte
- * that would otherwise have been returned by the next call to {@link #nextByte()}.
+ * that would otherwise have been returned by the next call to {@link #getByte()}.
  */
-public interface ByteIterator {
+public interface ByteCursor {
     /**
-     * Check if the iterator has at least 8 bytes left.
-     * Note that when this method returns {@code false}, the {@link #hasNextByte()} can still return {@code true}.
+     * Check if the iterator has at least 8 bytes left, and if so, read those 8 bytes and move the cursor forward.
+     * The bytes are packed as a {@code long} value in big-endian format, such that the highest-order byte
+     * in the long, is the byte that would otherwise have been returned by the next call to {@link #getByte()},
+     * after a call to {@link #nextByte()}.
+     * The bytes (as a {@code long}) will then be available through the {@link #getLong()} method.
+     * <p>
+     * Note that when this method returns {@code false}, the {@link #nextByte()} can still return {@code true}.
      * It is recommended to have any long-processing loop be followed by a byte-processing loop for the 7 or fewer
-     * bytes that might form a tail in the iterator.
+     * bytes that might form a tail in the cursor.
+     * <p>
+     * Also note that this method will not influence what is returned the {@link #getByte()} method.
      *
-     * @return {@code true} if a call to {@link #nextLong()} would succeed, otherwise {@code false}.
+     * @return {@code true} if the cursor read 8 bytes and moved forward, otherwise {@code false}.
      */
-    boolean hasNextLong();
+    boolean nextLong();
```

Comment:
I wonder if we should rename this to make it more clear that exactly happens here (Like moving the cursor forward).

Reply:
Hmm… `readLong`?

Reply:
yeah maybe... its just a bit odd in the sense that `readLong` method on the `ByteBuf` actually returns a long ..